src/snsary/contrib/octopus.py

```diff
@@ -17,38 +24,50 @@
 
 
 class OctopusSensor(PollingSensor):
-    CONSUMPTION_URL = "https://api.octopus.energy/v1/electricity-meter-points/{mpan}/meters/{serial_number}/consumption/?period_from={period_from}&order_by=period"
+    CONSUMPTION_URL = "https://api.octopus.energy/v1/{fuel_type}-meter-points/{mpxn}/meters/{serial_number}/consumption/?period_from={period_from}&order_by=period"
+
+    @classmethod
+    def electricity_from_env(cls):
+        return cls(
+            mpxn=os.environ["OCTOPUS_ELECTRICITY_MPAN"],
+            serial_number=os.environ["OCTOPUS_ELECTRICITY_SERIAL"],
+            token=os.environ["OCTOPUS_TOKEN"],
+            fuel_type="electricity",
+        )
 
     @classmethod
-    def from_env(cls):
+    def gas_from_env(cls):
         return cls(
-            mpan=os.environ["OCTOPUS_MPAN"],
-            serial_number=os.environ["OCTOPUS_SERIAL"],
+            mpxn=os.environ["OCTOPUS_GAS_MPRN"],
+            serial_number=os.environ["OCTOPUS_GAS_SERIAL"],
             token=os.environ["OCTOPUS_TOKEN"],
+            fuel_type="gas",
         )
 
-    def __init__(self, *, mpan, serial_number, token):
+    def __init__(self, *, mpxn, serial_number, token, fuel_type):
         PollingSensor.__init__(
             self,
             period_seconds=30 * 60,  # 30 mins
         )
 
         self.__token = token
-        self.__mpan = mpan
+        self.__mpxn = mpxn
         self.__serial_number = serial_number
+        self.__fuel_type = fuel_type
 
     @property
     def name(self):
-        return "octopus"
+        return f"octopus.{self.__fuel_type}"
```

Comment:
This doesn't allow for multiple meters per meter point, which should be a niche enough scenario that it's not worth making the names more complex to support it.